config/configsanity.go

```diff
@@ -1094,6 +1094,9 @@ func (cfg *CGRConfig) checkConfigSanity() error {
 			return fmt.Errorf("<%s> the CleanupInterval needs to be bigger than 0", utils.AnalyzerS)
 		}
 	}
+	if err := cfg.prometheusAgentCfg.validate(cfg); err != nil {
```

Comment:
Please redesign validate since it is taking the main object as argument which is twisted.